internal/provider/resource_environment.go

```diff
@@ -329,9 +350,9 @@ func resourceEnvironmentCreate(ctx context.Context, d *schema.ResourceData, meta
 		ErrorLog.Printf("Job Polling failed but continuing with env creation. Error: %v", job_err)
 	}
 
-	if job_status == Failed {
+	if job_status == Failed || job_status == Canceled || job_status == Abandoned {
```

Comment:
let's create a helper function instead of repeating

Reply:
done